base/client.jl

```diff
@@ -317,6 +319,21 @@ function __atreplinit(repl)
 end
 _atreplinit(repl) = invokelatest(__atreplinit, repl)
 
+"""
+    after_repl_init(f)
+
+Like [`atreplinit`](@ref) but calls `f` with `Base.active_repl` immediately if REPL is
+already initialized.
+"""
+function after_repl_init(f)
+    if isdefined(Base, :active_repl)
+        f(active_repl)
+    else
+        atreplinit(f)
```

Comment:
You're effectively implementing "maybe before, maybe after," while the function name here implies that `f` will run only once `repl` is fully up and running. If this branch is taken, `f` will run before `activate_repl` has defined its channels (see the implementation of `run_main_repl` in client.jl and `run_repl` in the stdlib).